code/game/machinery/vending/cm_vending.dm

```diff
@@ -854,6 +854,10 @@ GLOBAL_LIST_EMPTY(vending_products)
 		stock(I, user)
 
 /obj/structure/machinery/cm_vending/sorted/proc/stock(obj/item/item_to_stock, mob/user)
+	if (istype(src, /obj/structure/machinery/cm_vending/sorted/medical))
+		to_chat(user, SPAN_WARNING("[src] cannot be restocked!"))
+		return
+
```

Comment:
`
/obj/structure/machinery/cm_vending/sorted/medical/stock(obj/item/item_to_stock, mob/user)
		to_chat(user, SPAN_WARNING("[src] cannot be restocked!"))
		return
`
Don't typecheck on the root object, just put the /stock(obj/item/item_to_stock, mob/user) proc onto /obj/structure/machinery/cm_vending/sorted/medical like above, next to wherever /obj/structure/machinery/cm_vending/sorted/medical is defined.

Reply:
Yeah, I was having issues overriding the proc because I didn't realize I needed to remove proc in the child definition. Should be fixed.